orchagent/srv6orch.cpp

```diff
@@ -89,6 +95,222 @@ static bool mySidDscpModeToSai(const string& mode, sai_tunnel_dscp_mode_t& sai_m
+    Orch(tables),
+    m_vrfOrch(vrfOrch),
+    m_switchOrch(switchOrch),
+    m_neighOrch(neighOrch),
+    m_sidTable(applDb, APP_SRV6_SID_LIST_TABLE_NAME),
+    m_mysidTable(applDb, APP_SRV6_MY_SID_TABLE_NAME),
+    m_piccontextTable(applDb, APP_PIC_CONTEXT_TABLE_NAME),
+    m_mysidCfgTable(cfgDb, CFG_SRV6_MY_SID_TABLE_NAME),
+    m_locatorCfgTable(cfgDb, CFG_SRV6_MY_LOCATOR_TABLE_NAME),
+    m_counter_manager(SRV6_STAT_COUNTER_FLEX_COUNTER_GROUP, StatsMode::READ, SRV6_STAT_COUNTER_POLLING_INTERVAL_MS, false)
+{
+    m_neighOrch->attach(this);
+
+    initializeCounters();
+}
+
+Srv6Orch::~Srv6Orch()
+{
+    m_neighOrch->detach(this);
+}
+
+void Srv6Orch::initializeCounters()
+{
+    m_mysid_counters_supported = queryMySidCountersCapability();
+    if (!m_mysid_counters_supported)
+    {
+        SWSS_LOG_INFO("SRv6 counters are not supported on this platform");
+        return;
+    }
+
+    m_asic_db = make_shared<DBConnector>("ASIC_DB", 0);
+    m_counter_db = make_shared<DBConnector>("COUNTERS_DB", 0);
+    m_mysid_counters_table = make_unique<Table>(m_counter_db.get(), COUNTERS_SRV6_NAME_MAP);
+
+    if (gTraditionalFlexCounter)
+    {
+        m_vid_to_rid_table = make_unique<Table>(m_asic_db.get(), "VIDTORID");
+    }
+
+    m_counter_update_timer = new SelectableTimer(timespec { .tv_sec = SRV6_FLEX_COUNTER_UPDATE_TIMER , .tv_nsec = 0 });
+    auto et = new ExecutableTimer(m_counter_update_timer, this, "SRV6_FLEX_COUNTER_UPDATE_TIMER");
+    Orch::addExecutor(et);
+}
+
+bool Srv6Orch::queryMySidCountersCapability() const
+{
+    sai_attr_capability_t capability;
+    sai_status_t status = sai_query_attribute_capability(gSwitchId, SAI_OBJECT_TYPE_MY_SID_ENTRY, SAI_MY_SID_ENTRY_ATTR_COUNTER_ID, &capability);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Could not query SRv6 MySID entry attribute SAI_MY_SID_ENTRY_ATTR_COUNTER_ID %d", status);
+        return false;
+    }
+
+    return capability.set_implemented && capability.create_implemented;
+}
+
+
+bool Srv6Orch::getMySidCountersEnabled() const
+{
+    return m_mysid_counters_enabled;
+}
+
+bool Srv6Orch::getMySidCountersSupported() const
+{
+    return m_mysid_counters_supported;
+}
+
+IpAddress Srv6Orch::getMySidAddress(const sai_my_sid_entry_t& sai_entry) const
+{
+    ip_addr_t ip_addr = {};
+    ip_addr.family = AF_INET6;
+    memcpy(&ip_addr.ip_addr.ipv6_addr, sai_entry.sid, sizeof(ip_addr.ip_addr.ipv6_addr));
+
+    return IpAddress(ip_addr);
+}
+
+string Srv6Orch::getMySidCounterKey(const sai_my_sid_entry_t& sai_entry) const
+{
+    auto mysid_addr = getMySidAddress(sai_entry).to_string();
+    auto locator_cfg = getMySidEntryLocatorCfg(sai_entry);
+    return getMySidPrefix(mysid_addr, locator_cfg);
+}
+
+bool Srv6Orch::addMySidCounter(const sai_my_sid_entry_t& sai_entry, sai_object_id_t& counter_oid)
```

Comment:
Looks like we are not updating sai_object_id_t, i think we don't need to pass it as reference `sai_object_id_t& counter_oid`